actionpack/lib/action_dispatch/testing/assertions/response.rb

```diff
@@ -85,17 +86,13 @@ def normalize_argument_to_redirection(fragment)
           end
         end
 
-        def generate_response_message(type)
-          message = "Expected response to be a <#{type}>, but was"
-
+        def response_message
           if @response.redirection?
             redirect_is = normalize_argument_to_redirection(@response.location)
-            message << " a redirect to <#{redirect_is}>"
+            "<302> redirected to <#{redirect_is}>"
```

Comment:
The `302` could be replaced with `Rack::Utils::SYMBOL_TO_STATUS_CODE[:found]`. Let me know if you think that might be better.